types/dygraphs/dygraphs-tests.ts

```diff
@@ -1,611 +1,278 @@
-            eventDiv.innerHTML += "dblclick: " + nameAnnotation(ann) + "<br/>";
-        },
-        annotationMouseOverHandler: function(ann, point, dg, event) {
-            document.getElementById(nameAnnotation(ann)).style.fontWeight = 'bold';
-            saveBg = ann.div.style.backgroundColor;
-            ann.div.style.backgroundColor = '#ddd';
         },
-        annotationMouseOutHandler: function(ann, point, dg, event) {
-            document.getElementById(nameAnnotation(ann)).style.fontWeight = 'normal';
-            ann.div.style.backgroundColor = saveBg;
+    },
+    drawPoints: true,
+    errorBars: true,
+    height: 320,
+    highlightCircleSize: 2,
+    highlightSeriesOpts: {
+        highlightCircleSize: 5,
+        strokeBorderWidth: 1,
+        strokeWidth: 3,
+    },
+    labels: ['Date', 'Y1', 'Y2', 'Y3', 'Y4'],
+    labelsDiv: new HTMLElement(),
+    labelsKMB: true,
+    labelsSeparateLines: true,
+    legend: 'always',
+    rollPeriod: 14,
+    series: {
+        y: {
+            strokePattern: [25, 5],
+            strokeWidth: 3,
+            highlightCircleSize: 10,
+            drawPoints: true,
+            pointSize: 1.5,
+            axis: 'y2',
         },
+    },
+    showRoller: true,
+    stackedGraph: false,
+    strokeBorderWidth: 1,
+    strokeWidth: 2,
+    width: 480,
+    y2label: 'Secondary y-axis',
+    ylabel: 'Primary y-axis',
+    underlayCallback: (canvas, area, dygraph) => {
+        // $ExpectType CanvasRenderingContext2D
+        canvas;
+
+        // $ExpectType Readonly<Area>
+        area;
+
+        // $ExpectType Readonly<Dygraph<HTMLDivElement>>
+        dygraph;
+    },
+    drawCallback: (dygraph, is_initial) => {
+        // $ExpectType Readonly<Dygraph<HTMLDivElement>>
+        dygraph;
+
+        // $ExpectType boolean
+        is_initial;
+    },
+    highlightCallback: (event, x, points, row, seriesName) => {
+        // $ExpectType MouseEvent
+        event;
+
+        // $ExpectType number
+        x;
+
+        // $ExpectType ReadonlyArray<Point<HTMLDivElement>>
+        points;
+
+        // $ExpectType number
+        row;
+
+        // $ExpectType string
+        seriesName;
+    },
+    unhighlightCallback: event => {
+        // $ExpectType MouseEvent
+        event;
+    },
+    clickCallback: (event, x, points) => {
+        // $ExpectType MouseEvent
+        event;
+
+        // $ExpectType number
+        x;
+
+        // $ExpectType ReadonlyArray<Point<HTMLDivElement>>
+        points;
+    },
+    pointClickCallback: (event, point) => {
+        // $ExpectType MouseEvent
+        event;
+
+        // $ExpectType Readonly<Point<HTMLDivElement>>
+        point;
+    },
+    zoomCallback: (minDate, maxDate, yRanges) => {
+        // $ExpectType number
+        minDate;
+
+        // $ExpectType number
+        maxDate;
+
+        // $ExpectType ReadonlyArray<[number, number]>
+        yRanges;
+    },
+});
+
+// $ExpectType void
+d.adjustRoll(10);
+
+// $ExpectType 10
```

Comment:
This is incorrect. The return type should be `number`. There's no reason to expect `toDataXCoord` to be the identity function. (Also for several methods below.)

Reply:
Thanks, fixed